compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2965,6 +2979,190 @@ static void generateArrayElementLoad(TR::Node* node, TR::Register* valueReg, uin
    generateRegMemInstruction(loadOpCode,  node, valueReg, generateX86MemoryReference(addressReg, index, cg), cg);
    }
 
+/** \brief
+*    Generate instructions to do array copy with copy size as a constant
+*
+*  \param node
+*     The tree node
+*
+*  \param dstReg
+*     The destination array address register
+*
+*  \param srcReg
+*     The source array address register
+*
+*  \param sizeReg
+*     The register holding the total size of elements to be copied, in bytes
+*
+*  \param elementSize
+*     The size of an element, in bytes
+*
+*  \param copySize
+*     The copy size, in bytes
+*
+*  \param cg
+*     The code generator
+*/
+static void arrayCopyPrimitiveInlineSmallSizeConstantCopySize(TR::Node* node,
+                                                              TR::Register* dstReg,
+                                                              TR::Register* srcReg,
+                                                              TR::Register* sizeReg,
+                                                              uint32_t copySize,
+                                                              TR::CodeGenerator* cg)
+   {
+   if (cg->comp()->getOption(TR_TraceCG))
+      {
+      traceMsg(cg->comp(), "%s: node n%dn srcReg %s dstReg %s sizeReg %s copySize %d\n", __FUNCTION__, node->getGlobalIndex(),
+         cg->comp()->getDebug()->getName(srcReg), cg->comp()->getDebug()->getName(dstReg), cg->comp()->getDebug()->getName(sizeReg), copySize);
+      }
+
+   if (copySize == 0)
+      return;
+
+   TR::Register* tmpReg1 = cg->allocateRegister(TR_GPR);
+   TR::Register* tmpReg2 = cg->allocateRegister(TR_GPR);
+   TR::Register* tmpVRFReg1 = cg->allocateRegister(TR_VRF);
+   TR::Register* tmpVRFReg2 = cg->allocateRegister(TR_VRF);
+
+   if ((copySize == 1) || (copySize == 2) || (copySize == 4) || (copySize == 8) || (copySize == 16) || (copySize == 32)
```

Comment:
If you know copySize is non-zero positive, you should look into whether a power of two test is cheaper than this `if` expression.  e.g., something like:
```
   if ((copySize <= 32) && ((copySize & (copySize-1)) == 0))
      || ((copySize == 64) && cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_X86_AVX512F)))
```
There's a bunch of bit tests in infra/Bit.hpp for this kind of check, but I think you can do better if you know `copySize` is non-zero and positive already.

Reply:
Addressed  in [d2c22d0](https://github.com/eclipse/omr/compare/d2c22d0bf1d27788f27712939d4e22e79ba06e0c..24c7d3f9ddf90cc7165e3bacb2ca8d63d053c0d7)